### DISCUSSION TITLE
How can I handle multiple queries with useInfinityQuery?

### DISCUSSION THREAD

**Discussion by alioshr** [TanStack/query#4806] [Q&A] [answered]
I have an use case that consists in loading multiple queries, which is currently handled by useQueries. All good with that. The thing is that now the BE will return data in chunks using a cursor and I need to load all chunks of data in case we have cursors. 

It would be something like using useQueries, but each query needs to support the functionality of loading more chunks of data while cursors are available. 

I have looked into the documentation and found nothing about this functionality. 

As far as I know useInfiniteQuery just supports one query and useQueries does not support to fetch more chunks of data as with useInfiniteQuery. 

Thanks in advance! I hope I have managed to explain this properly =D

**Comment by TkDodo** [comment, marked as answer]
there is no `useQueries` for infinite queries, sorry.

**Reply by alioshr** [reply]
Alright. I did a recursion within the api call function, which gets passed into the query fn param. I guess this is the best that we can do right now. And it worked for me because in this use case the user does not request more data. It was just a strategy to remove load from the api calls when having too many results.

Thanks anyway

**Reply by anatoliilikutin** [reply]
Hi @TkDodo, is there a specific reason for not having useQueries for infinite queries? It would be quite helpful in our case for Smart TV project where we have rails of data, and each rail can be paginated

**Reply by TkDodo** [reply]
too complex with too little gains. I would just map over the result data and render a single component with _one_ infinite query each. Having one component that renders multiple, dynamic infinite scrolling queries seems like a lot.

**Reply by perenstrom** [reply]
Just wanna chime in and say that I have come to a point where I would need this as well. Ironically enough to integrate with Tanstack Table. I have a server side grouped tanstack table with lazy loading of leaf items in expanded groups, and in those I need paginated results.
